### PR TITLE
Resolve unknown xref as fallback

### PR DESCRIPTION
**Pull request by yufeih**
*edge-developer* content contains `@font-face` that are incorrectly parsed as xref. This PR lefts unknown xrefs as is and report an `xref-not-found` warning.

#2782

### CODE REVIEW

File: docs/specs/build/xref.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+# Unknown xrefs are output as is with a warning
+inputs:
+  docfx.yml:
+  docs/a.md: Link to @xref
+outputs:
+  docs/a.json: |
+    { "content": "<p>Link to @xref</p>" }
+  build.log: |
+    ["warning","xref-not-found","Cannot find uid 'xref' using xref '@xref'","docs/a.md"]
```
Comment (superyyrrzz) [inline docs/specs/build/xref.yml]: @xref needn't warning as it's prone to false positive

Comment (yufeih) [reply docs/specs/build/xref.yml]: Does <xref> syntax show warnings?

File: src/docfx/lib/markdown/ResolveXref.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            {
+                document.Replace(node =>
+                {
+                    if (node is XrefInline xref)
+                    {
+                        var href = resolveXref(xref.Href);
+                        if (string.IsNullOrEmpty(href))
+                        {
+                            var raw = xref.GetAttributes().Properties.First(p => p.Key == "data-raw-source").Value;
+                            var error = raw.StartsWith("@")
```
Comment (superyyrrzz) [inline src/docfx/lib/markdown/ResolveXref.cs]: >var error = raw.StartsWith("@") [](start = 28, length = 31)

can judge from "data-throw-if-not-resolved"="true" to log warning

Comment (yufeih) [reply src/docfx/lib/markdown/ResolveXref.cs]: didn't know that. I think looking up html attributes for both `raw` and `should-throw `is a hack. I'll add them to markdig ast as part of the reference work.

**Pull request merged**